### PR TITLE
Introduce ethereum-types package

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

This PR moves general ethereum-related types from `@0xproject/types` to `ethereum-types`.

## Motivation and Context

In order to submit some of our typing (like web3) to DT, we need to add their dependencies to https://github.com/Microsoft/types-publisher/blob/master/dependenciesWhitelist.txt

It needs to be generic and reusable enough for that.
This package just contains a bunch of general ethereum types to be reused between ethereum libs later.

## How Has This Been Tested?

It builds

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

**Comment by fabioberger** [comment]
This PR is going to break the docs the rely on displaying type defs now in `ethereum-types`. Please add `../ethereum-types/src/index.ts`, to the `extraFileIncludes` section in `configs.docPublishConfigs` section of `package.json` of packages we generate docs for.

**Comment by LogvinovLeon** [comment]
@fabioberger Fixed!

File: packages/ethereum-types/CHANGELOG.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+    {
+        "version": "0.0.1",
+        "changes": [
+            {
+                "note": "Initial publish"
+            }
```
Comment (fabioberger) [inline packages/ethereum-types/CHANGELOG.json]: Add PR number.

File: packages/ethereum-types/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    "license": "Apache-2.0",
+    "repository": {
+        "type": "git",
+        "url": "https://github.com/0xProject/0x-monorepo.git"
+    },
+    "bugs": {
+        "url": "https://github.com/0xProject/0x-monorepo/issues"
+    },
+    "homepage": "https://github.com/0xProject/0x-monorepo/packages/types/README.md",
```
Comment (fabioberger) [inline packages/ethereum-types/package.json]: Fix URL. `ethereum-types`, not `types`.

**Pull request merged**